### ISSUE TITLE
binutls: package contains a newline

### ISSUE THREAD

**Issue by vsoch** [LLNL/spack#31779]
```
==> binutils: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16' 'V=1'

1 error found in build log:
     113    configure: WARNING: cache variable ac_cv_prog_LD contains a newline
     114    configure: creating ./config.status
     115    config.status: creating Makefile
     116    configure: WARNING: cache variable ac_cv_prog_LD contains a newline
     117    ==> binutils: Executing phase: 'build'
     118    ==> [2022-07-27-12:16:15.758440] 'make' '-j16' 'V=1'
  >> 119    Makefile:399: *** target pattern contains no '%'.  Stop.
```
To reproduce how I'm doing it:

```bash
$ singularity pull docker://ghcr.io/buildsi/spliced-experiment
$ singularity exec --home $PWD --bind /p/vast1/build/experiment/cache:/cache --bind /p/vast1/build/experiment/spack-opt:/spack/opt /p/vast1/build/experiment/spliced-experiment_latest.sif spliced splice --package tau@2.26.3 --splice libunwind --runner spack --replace libunwind --experiment tau --outfile /p/vast1/build/experiment/results/tau/2.26.3/libunwind/experiment.json
```

**Comment by haampie** [comment]
This seems to be caused by `spliced` given that

```
spack install -v binutils@2.38 ~gas~gold+headers~interwork~ld+libiberty~lto+nls+plugins libs=shared,static 
```

works fine in the container.

**Comment by vsoch** [comment]
How exactly?

**Comment by vsoch** [comment]
Also this container has changed like 20 times since I posted the issue.

**Comment by vsoch** [comment]
Also spliced (despite it's name) is not actually doing the splicing - it's calling spack functions, so this is still an issue with spack. I'll use the updated container today to try and reproduce for you something in Python to cause it.

**Comment by haampie** [comment]
I could reproduce in docker by copying your command, but running spack install directly worked. However, it takes forever to reproduce, so if you're sure this issue is in spack, then please make the reproducer smaller...

Does `spliced` set `LD=ld\n` perhaps?

**Comment by vsoch** [comment]
Yeah I can try to do that. And lolz I package my entire environment to reproduce which is more than anyone ever does and it’s still not good enough… 😭

**Comment by haampie** [comment]
It's good to have a proper reproducer, but it's still asking a lot of time from others to reproduce...

Fyi, this looks similar:

```
$ spack stage binutils
$ cd /tmp/whereever/spack/puts/it/spack-src
$ LD=$'\nld' sh -c './configure > /dev/null && make'
configure: WARNING: cache variable ac_cv_env_LD_value contains a newline
configure: WARNING: cache variable ac_cv_prog_LD contains a newline
configure: WARNING: cache variable ac_cv_env_LD_value contains a newline
configure: WARNING: cache variable ac_cv_prog_LD contains a newline
Makefile:399: *** missing separator.  Stop.
```

so my suspicion is that your wrapper around spack sets some variable that affects the build. I don't think Spack controls the `LD` variable

**Comment by vsoch** [comment]
The wrapper doesn’t set any variables that aren’t already set in the container. And if that were the case I’d see it across packages (but I don’t).

**Comment by vsoch** [comment]
And next time I won’t provide a container I guess!

**Comment by vsoch** [comment]
okay I found the difference - for a spack install (or spack python) it does it's own logic to setup paths / environment. For spliced when we add spack to the path, we just do those steps, here:

https://github.com/buildsi/spliced/blob/da091ab8822261ea1059e9620ab1b89c27f81f95/spliced/utils/terminal.py#L94-L108

so the real question is what other foo is spack doing on init?

**Comment by vsoch** [comment]
So I think what I need to do is run spliced from spack python, which also needs the full path to the "executable" (which is just a Python script entrypoint).

```bash
$ spack python /usr/local/bin/spliced splice --package tau@2.23.1 --splice hwloc --runner spack --replace hwloc --experiment tau --outfile /results/tau/2.23.1/hwloc/experiment.json
```

**Comment by vsoch** [comment]
TLDR - spack doesn't work like a typical Python module. Thanks for the help @haampie !
